File: src/main/java/it/unical/demacs/enterprise/fintedapp/dto/PostBasicInfoDto.java
package it.unical.demacs.enterprise.fintedapp.dto;

import java.util.Date;

import it.unical.demacs.enterprise.fintedapp.data.entities.PostStatus;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
@ToString
@SuppressWarnings(value = {"unused"})
public class PostBasicInfoDto {
	private Long id;

	@NotEmpty
	private String title;
	
	@NotNull
	private Long sellerId;
	
	@NotNull
	private String sellerUsername;
	
	private Date publishedDate;
	
	@PositiveOrZero
	private Long startingPrice;
	
	private PostStatus status;
}

File: FINTed_android/app/src/main/java/it/unical/demacs/enterprise/fintedapp/models/PostBasicInfoDto.kt
package it.unical.demacs.enterprise.fintedapp.models

/**
 *
 * @param id
 * @param title
 * @param sellerId
 * @param sellerUsername
 * @param publishedDate
 * @param startingPrice
 * @param status
 */

data class PostBasicInfoDto(
    val id: kotlin.Long? = null,
    val title: kotlin.String? = null,
    val sellerId: kotlin.Long? = null,
    val sellerUsername: kotlin.String? = null,
    val publishedDate: java.util.Date? = null,
    val startingPrice: kotlin.Long? = null,
    val status: PostDto.Status? = null
)
